bitcoin/test/fuzz/miniscript_random.cpp

```diff
@@ -402,6 +402,17 @@ FUZZ_TARGET_INIT(miniscript_random, initialize_miniscript_random)
     assert(decoded->ToScript(PARSER_CTX) == script);
     assert(decoded->GetType() == node->GetType());
 
+    if (fuzzed_data_provider.ConsumeBool()) {
+        // Optionally pad the script with OP_NOPs to max op the ops limit of the constructed script.
+        // This makes the script obviously not actually miniscript-compatible anymore, but the
+        // signatures constructed in this test don't commit to the script anyway, so the same
+        // miniscript satisfier will work. This increases the sensitivity of the test to the ops
+        // counting logic being too low, especially for simple scripts.
+        for (int i = node->GetOps(); i < MAX_OPS_PER_SCRIPT; ++i) {
+            script.push_back(OP_NOP);
+        }
+    }
```

Comment:
Wouldn't it be preferable to have it at the "seed level"? That is having larger seeds; for instance by running the target with a large `-max_len` you would maximize the number of OPs.

Reply:
That's not comparable, I think. By stuffing the OP_NOPs in, you can get failures instantly whenever there is an overestimate, with the simplest script construction that triggers it, as input. Relying on large inputs means the fuzzer needs to get a large number of choices right to end up with a sufficiently large script that exploits it.

Of course, this isn't *needed* - but I think it makes the test significantly more sensitive to ops counting errors.

Reply:
Ok, i can't think of any case where padding with NOPs would hide anything below.